jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -895,17 +970,18 @@ public Mutable path(String path)
             if (!URIUtil.isPathValid(path))
                 throw new IllegalArgumentException("Path not correctly encoded: " + path);
             _uri = null;
-            _path = path;
+            _path = null;
             _canonicalPath = null;
+            String param = _param;
+            _param = null;
+            parse(State.PATH, path);
```

Comment:
I'm cautious about this, as this now allows for a query and fragment to be passed in via this method.  Can this be done without a call to parse?  perhaps we should check for ? and # characters before parsing... or does the isPathValid already do that?
Note we should javadoc if the path is raw or encoded.

Reply:
There's a fair number of places that uses this `HttpURI.Mutable.path(String)` API.
Interestingly, we actually need to parse the path in some of those places to satisfy the Servlet 6.0 "suspicious characters" rule. (eg: `sendRedirect`, `getRequestDispatcher`, etc)
I'm not sure what `isPathValid` you are mentioning here.
It would be great if we had a `parsePath` that only did the path, and if it encountered something outside of the scope of PATH (eg: query / fragment / etc) that would also be an error.

Reply:
> Note we should javadoc if the path is raw or encoded.

As far as I'm concerned, this behavior didn't change for `HttpURI.Mutable.path(String)`.
But since it wasn't originally documented, I'm not sure the intent of this method.

Reply:
@gregw bump

Reply:
@joakime I've pushed a fix from my concern.... if you don't like it, we can revert and I'll OK this anyway.